python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,64 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids. This inherited class adds
+    the removal of saturated, interpolated, and edge_key objects to the set of
+    bad flags. It is a temporary addition designed preserve the source selction
+    used in matchOptimisticB. Once matchPessimisticB is addopted as the default
+    source selector the class will be removed and the saturated, interpoalted, and
+    edge_key flags will be added to the matcherSourceSelector class.
```

Comment:
This is clear. I am not convinced others will be willing to abandon the old matcher as quickly as you would like (so I am not entirely happy with the class name for this star selector), but at least this matcher provides a useful separation.

Since replacing the old matcher with this one is technical debt, please file a ticket to replace the old selector (and matcher) and turn this into a TODO note that references the ticket number.

"interpoalted"